addons/mail/static/src/core/common/composer.js

```diff
@@ -763,18 +766,30 @@ export class Composer extends Component {
             Object.assign(config, {
                 emailAddSignature: true,
                 text: composer.text,
+                mentionedPartners: composer.mentionedPartners.map(
+                    (p) => ({ id: p.id, type: p.type })
+                ),
+                mentionedChannels: composer.mentionedChannels.map(
+                    (c) =>  ({ id: c.id, model: c.model })
+                ),
             });
         }
         browser.localStorage.setItem(composer.localId, JSON.stringify(config));
     }
 
-    restoreContent() {
+    async restoreContent() {
         const composer = toRaw(this.props.composer);
         try {
             const config = JSON.parse(browser.localStorage.getItem(composer.localId));
             if (config.text) {
                 composer.emailAddSignature = config.emailAddSignature;
                 composer.text = config.text;
+                for (const mentionedPartner of config.mentionedPartners) {
+                    this.props.composer.mentionedPartners.push(this.store.Persona.get(mentionedPartner));
+                }
+                for (const mentionedChannel of config.mentionedChannels) {
+                    this.props.composer.mentionedChannels.push(await this.store.Thread.getOrFetch(mentionedChannel));
+                }
```

Comment:
We don't necessarily have local knowledge of the partner when restoring the content.

Reply:
Code should also be more defensive: the channel or partner may no longer exist anymore, or local data is outdated.

Reply:
> We don't necessarily have local knowledge of the partner when restoring the content.

Maybe but we don't have a way to fetch them yet…

> Code should also be more defensive: the channel or partner may no longer exist anymore, or local data is outdated.

I don't get what you want from me here if the local data are outdated and not updated the mention is going to work like any other previous mention.

Reply:
That works, yes

Reply:
I disagree here, we should not use insert based on "user input" or unreliable/untrusted data.

Only data validated by the server should be inserted. If we need data about a persona we don't have, we should fetch it appropriately.

Reply:
You have to create the route to fetch the given partner ids in the context of the current thread, and it should have similar rules/access check as the original route returning suggestions (in particular taking account channel restriction and project override).